erts/emulator/beam/bif.c

```diff
@@ -3440,16 +3442,26 @@ static int do_float_to_charbuf(Process *p, Eterm efloat, Eterm list,
                         continue;
                 }
             }
+        } else if (arg == am_short) {
+            fmt_type = FMT_SHORT;
+            continue;
         }
         goto badarg;
     }
+
     if (is_not_nil(list)) {
         goto badarg;
     }
 
     GET_DOUBLE(efloat, f);
 
-    if (fmt_type == FMT_FIXED) {
+    if (fmt_type == FMT_SHORT) {
+        const int index = d2s_buffered_n(f.fd, fbuf);
+
+        // Terminate the string.
```

Comment:
I think you should replace this comment with a /*...*/ comment so the coding style is kept consistent. Line comments are not used anywhere else in OTP's C code. I have seen that Ulf Adams uses line comments in his code that is included in this PR but I think it is better to keep that code as it is to make it easier to patch if/when he updates his repository.